ocaml/typing/printtyp.ml

```diff
@@ -1125,6 +1125,9 @@ let rec tree_of_typexp mode ty =
             | Tconstr(path, [ty], _)
               when Path.same path Predef.path_option ->
                 tree_of_typexp mode ty
+            | Tconstr(path, [], _)
+              when Path.same path Predef.path_lexing_position ->
+                Otyp_constr (Oide_ident (Out_name.create "lexing_position"), [])
```

Comment:
I think I would edit the `Otyp_arrow` constructor to contain label information. There is already some gross-ness around the `string` stored there -- when it's empty, there is no label and the `:` is dropped. If it's non-empty, the label is printed. And for an optional argument, the label begins with `?`. All pretty bad. Perhaps `Otyp_arrow` would be better just storing a label directly (not as a `string`)? In any case, that's how I would proceed here, because we don't actually want to print `lexing_position` in this case.

Reply:
That makes sense. I think I still need to construct some Otyp here though, unless I change the `Otyp_arrow` constructor to store an `out_type option` for the argument type instead. 
Should I keep the hacky Otyp construction, make the type optional, or something else entirely? Not sure which is worse

Reply:
We’d rather have complete garbage than mildly reasonable garbage. Let’s just chuck a unit there and base the printing based on the Position alone

Reply:
I have pushed a very ugly implementation of printing *without* relying on creating a nontrivial Outcometree node. Once the new PR for adding label information to `Otyp_arrow` is in, the result will be nicer.